hnn_core/cell.py

```diff
@@ -137,26 +141,134 @@ def gid(self, gid):
+    def _set_biophysics(self, p_secs):
+        "Set the biophysics for the default Pyramidal cell."
+
+        # neuron syntax is used to set values for mechanisms
+        # sec.gbar_mech = x sets value of gbar for mech to x for all segs
+        # in a section. This method is significantly faster than using
+        # a for loop to iterate over all segments to set mech values
+
+        # Value depends on distance from the soma. Soma is set as
+        # origin by passing cell.soma as a sec argument to h.distance()
+        # Then iterate over segment nodes of dendritic sections
+        # and set gbar_ar depending on h.distance(seg.x), which returns
+        # distance from the soma to this point on the CURRENTLY ACCESSED
+        # SECTION!!!
+        h.distance(sec=self.soma)
+        for sec in self.sections:
+            sec_name = sec.name().split('_', 1)[1]
+            for mech_name, mech in p_secs[sec_name]['mechs'].items():
+                sec.insert(mech_name)
+                for attr, val in mech.items():
+                    if hasattr(val, '__call__'):
+                        sec.push()
+                        for seg in sec:
+                            setattr(seg, attr, val(h.distance(seg.x)))
+                        h.pop_section()
+                    else:
+                        setattr(sec, attr, val)
+
+    def _create_synapses(self, p_secs, p_syn):
+        """Create synapses."""
+        for sec_name in p_secs:
+            for receptor in p_secs[sec_name]['syns']:
+                sec_name_sanitized = sec_name.replace('_', '')
+                syn_key = f'{sec_name_sanitized}_{receptor}'
+                if sec_name == 'soma':
+                    seg = self.soma(0.5)
+                else:
+                    seg = self.dends[sec_name](0.5)
+                self.synapses[syn_key] = self.syn_create(
+                    seg, **p_syn[receptor])
+
+    def _create_sections(self, p_secs):
+        """Create soma and set geometry.
+
+        Notes
+        -----
+        By default neuron uses xy plane
+        for height and xz plane for depth. This is opposite for model as a
+        whole, but convention is followed in this function ease use of gui.
+        """
+        sec_pts, _, _, _, topology = self.secs
+
+        for sec_name in p_secs:
+            if sec_name == 'soma':
+                self.soma = h.Section(cell=self, name=self.name + '_soma')
+                sec = self.soma
+            else:
+                self.dends[sec_name] = h.Section(
+                    name=self.name + '_' + sec_name)
+                sec = self.dends[sec_name]
+
+            h.pt3dclear(sec=sec)
+            for pt in sec_pts[sec_name]:
+                h.pt3dadd(pt[0], pt[1], pt[2], 1, sec=sec)
+            sec.L = p_secs[sec_name]['L']
+            sec.diam = p_secs[sec_name]['diam']
+            sec.Ra = p_secs[sec_name]['Ra']
+            sec.cm = p_secs[sec_name]['cm']
+
+            if sec.L > 100.:
```

Comment:
Relevant for #150, what are the units here?

Reply:
it's um as in rest of Neuron. I added a tiny comment saying so.